src/com/seafile/seadroid2/ui/adapter/SeafItemAdapter.java

```diff
@@ -72,10 +86,23 @@ public boolean isEmpty() {
      * <p>
      * This method should be called after the download folder button was clicked.
      * 
-     * @param downloadTaskInfos
+     * @param newList
      */
-    public void setDownloadTaskList(List<DownloadTaskInfo> downloadTaskInfos) {
-        this.mDownloadTaskInfos = downloadTaskInfos;
+    public void setDownloadTaskList(List<DownloadTaskInfo> newList) {
+
+        if (this.mDownloadTaskInfos == null || newList.size() != this.mDownloadTaskInfos.size()) {
+            this.mDownloadTaskInfos = newList;
+            notifyDataSetChanged();
```

Comment:
The caller does not know if the new downloadTaskInfos actually contains changes, that require a UI refresh. Doing the notifyDataSetChanged() by the caller after every setDownloadTaskList() leads to repeated thumbnails fetches due to the timer (in case the the server could not provide the thumbnail for some reason).